src/DataSourceProvider.vue

```diff
@@ -147,17 +149,23 @@ export default {
     },
     onAddDefaultSecurity() {
       this.isLoading = true;
-      this.selectedDataSource.accessRules = this.widgetData.accessRules;
+
+      if (this.selectedDataSource.accessRules && this.selectedDataSource.accessRules.length > 0) {
```

Comment:
Why not just push them all in anyway? I don't think this `if` condition is adding much benefit or increase in performance.

Reply:
@tonytlwu old data sources do not have an `accessRules`. It can be a null or empty array. With an empty array, your suggestion will work but not with null.

Reply:
👍